modules/cudacodec/include/opencv2/cudacodec.hpp

```diff
@@ -380,59 +378,18 @@ class CV_EXPORTS_W VideoReader
     */
     CV_WRAP virtual FormatInfo format() const = 0;
 
-    /** @brief Grabs the next frame from the video source.
-
-    @param stream Stream for the asynchronous version.
-    @return `true` (non-zero) in the case of success.
-
-    The method/function grabs the next frame from video file or camera and returns true (non-zero) in
-    the case of success.
-
-    The primary use of the function is for reading both the encoded and decoded video data when rawMode is enabled.  With rawMode enabled
-    retrieve() can be called following grab() to retrieve all the data associated with the current video source since the last call to grab() or the creation of the VideoReader.
-     */
-    CV_WRAP virtual bool grab(Stream& stream = Stream::Null()) = 0;
```

Comment:
Why do you removed grab? It breaks compatibility and it's still mentioned in other methods.

Reply:
> now I am wondering if it is the right time to make some more alterations. Specifically those below which I "wrongly" introduced to mirror the VideoCapture class:
> 
>    - remove grab()/retrieve() for the decoded frame. This combination is completely unecessary grab() doesn't save any processing, it only remove the need to pass a GpuMat when you don't want to retrieve the decoded frame.
>    -  propertyVal of double in get(const VideoReaderProps propertyId, CV_OUT double& propertyVal) results in a lot of needless casting when the value will always be an integer.
> 
> What do you think?

I removed `grab()` in that commit so you could take a look.  It was an unecessary addition which adds a lot of code mess to `VideoCapture` and was "wrongly" introduced to mirror `VideoCapture` without actually mirroring the funcitonality.  I can back it out and place it in another PR if you think agree with its removal?

Reply:
@asmorkalov Added `grab()` back into to this PR.